pkg/dataobj/index/indexobj/builder.go

```diff
@@ -118,10 +119,10 @@ type Builder struct {
 
 	currentSizeEstimate int
 
-	builder       *dataobj.Builder // Inner builder for accumulating sections.
-	streams       *streams.Builder
-	pointers      *pointers.Builder
-	indexPointers *indexpointers.Builder
+	builder       *dataobj.Builder                                 // Inner builder for accumulating sections.
+	streams       map[multitenancy.TenantID]*streams.Builder       // The key is the TenantID.
```

Comment:
I don't feel you need `multitenancy.TenantID` as its own type. We can assume `string` is the tenant, especially with the comments. Consider the `logsobj.Builder`, which will also need its own `type TenantID string` type too. It gets messy quite quickly without providing any real benefits from a type safety perspective.